viewing_party/party.py

```diff
@@ -0,0 +1,162 @@
+def create_movie(title, genre, rating):
+    if title and genre and rating:
+        new_movie = {}
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+    else: 
+        new_movie = None
+    return new_movie
+    
+def add_to_watched(user_data, movie):
+    if movie:
+        user_data["watched"].append(movie)
+    else:
+        user_data["watched"]    
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie:
+        user_data["watchlist"].append(movie)
+    else:
+        user_data["watchlist"]
+    return user_data
+
+def watch_movie(user_data, title): 
+    for item in user_data["watchlist"]:   
+        if item["title"] == title:
+            user_data["watched"].append(item)
+    for item in user_data["watchlist"]:
+        if item in user_data["watched"]:
+            user_data["watchlist"].remove(item)
+    return user_data 
+
+def get_watched_avg_rating(user_data):
+    counter = 0
+    total = 0
+    if user_data["watched"] == []:
+        avg = 0.0
+    for item in user_data["watched"]:
+        if item["rating"]:
+            counter += 1
+            total += item["rating"]
+            avg = total/counter
+    return avg
+
+def get_most_watched_genre(user_data):
+    pass
+    if user_data["watched"] == []:
+        genre = None
+    counter = 1
+    for item in user_data["watched"]:
+        if item["genre"] == item["genre"]:
+            counter +=1
+            genre = item["genre"]
+    return genre
+        
+def get_unique_watched(user_data): # movies only user has watched
+    movies_friends_watched = set()
+    unique_movies = set()
+    list_unique_movies = []
+    user = user_data["watched"]
+    friends = user_data["friends"]
+    for friend in friends: # list of dictionaries
+        for movie in friend["watched"]:
+            movies_friends_watched.add(movie["title"])
+    for movie in user:
+        unique_movies.add(movie["title"])
+        for movie in movies_friends_watched:
+            if movie in unique_movies:
+                unique_movies.remove(movie)
+    list_unique_movies = [{"title": title} for title in unique_movies ] 
+    return list_unique_movies
+
+def get_friends_unique_watched(user_data):
+    user_watched = set()
+    movies_friends_watched = set()
+    list_movies = []# list of movies user watched
+    for movie in user_data["watched"]:
+        user_watched.add(movie["title"])
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            movies_friends_watched.add(movie["title"])
+            for movie in user_data["watched"]:
+                if movie["title"] in movies_friends_watched:
+                    movies_friends_watched.remove(movie["title"])
+    list_movies = [{"title" : title} for title in movies_friends_watched]
+    return list_movies
```

Comment:
Similar situation as `get_unique_watched()`. 

1) Instead of removing items from `movies_friends_watched` we can find the difference between the sets.

2) Instead of recreating the movie dictionary with only the title key, we can iterate over the user's watched list again and only append & return movies whose title is not in the `movies_friends_watched` set.